example/tensorflow/code_template/tensorflow_fashion_mnist.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras.datasets import fashion_mnist
+from tqdm import tqdm  # progress bars
+import numpy as np
+
+
+# convolution net initializator using low-level API of tensorflow
+def conv_net(x, keep_prob, train_shape):
+    shape_x = tf.reshape(x, [-1] + list(train_shape) + [1])
+
+    conv1_filter = tf.Variable(tf.truncated_normal(shape=[1, 1, 1, 64], mean=0, stddev=0.08))
+    conv2_filter = tf.Variable(tf.truncated_normal(shape=[1, 1, 64, 128], mean=0, stddev=0.08))
+
+    conv1 = tf.nn.conv2d(shape_x, conv1_filter, strides=[1, 1, 1, 1], padding='SAME')
+    conv1 = tf.nn.relu(conv1)
+    conv1_pool = tf.nn.max_pool(conv1, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
+    conv1_bn = tf.layers.batch_normalization(conv1_pool)
+
+    conv2 = tf.nn.conv2d(conv1_bn, conv2_filter, strides=[1, 1, 1, 1], padding='SAME')
+    conv2 = tf.nn.relu(conv2)
+    conv2_pool = tf.nn.max_pool(conv2, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
+    conv2_bn = tf.layers.batch_normalization(conv2_pool)
+
+    flat = tf.contrib.layers.flatten(conv2_bn)
+
+    dense = tf.layers.dense(flat, 128, activation=tf.nn.tanh)
+    dense = tf.nn.dropout(dense, keep_prob)
+    dense = tf.layers.batch_normalization(dense)
+
+    out = tf.layers.dense(dense, 10, activation=None)
+    return out
+
+
+# Train neural network over batch
+def train_neural_network(sess, optimizer, keep_probability, feature_batch, label_batch):
+    sess.run(optimizer,
+             feed_dict={
+                 x: feature_batch,
+                 y: label_batch,
+                 keep_prob: keep_probability
+             })
+
+
+# get loss over batch
+def get_batch_loss(sess, feature_batch, label_batch, cost):
+    return sess.run(cost,
+                    feed_dict={
+                        x: feature_batch,
+                        y: label_batch,
+                        keep_prob: 1.
+                    })
+
+
+# get accuracy over batch
+def get_batch_acc(sess, feature_batch, label_batch, accuracy):
+    return sess.run(accuracy,
+                    feed_dict={
+                        x: feature_batch,
+                        y: label_batch,
+                        keep_prob: 1.
+                    })
+
+
+# print stats of epoch
+def print_stats(epoch, train_losses, train_accs, valid_losses, valid_accs):
+    print((
+        'Epoch {} summary: '
+        'train loss: {:4}, train acc: {:4},'
+        'valid loss: {:4}, valid acc: {:4}'
+    ).format(
+        epoch,
+        train_losses.mean(), train_accs.mean(),
+        valid_losses.mean(), valid_accs.mean()
+    ))
+
+
+(train_x, train_y), (valid_x, valid_y) = fashion_mnist.load_data()
```

Comment:
@seventeen-iterator can you add the code below in methods too, so that it is easy to understand.

Reply:
Do you mean, all code after highlighted lines?

Reply:
Oh, you talk about print stats function. For some reason I could see this only in email.